File: leetcode/ValidParenth.java
import java.util.Stack;
import java.util.Iterator;
import java.util.Map;
import java.util.HashMap;
/**
 * check valid parenthesis
 */
public class ValidParenth {

    /**
     * @param testString string
     * @return boolean
     */
    public static boolean isValid(String testString) {
        Stack<Character> myStack = new Stack<>();
        //TODO: remember two parameter
        Map<Character, Character> myMap = new HashMap<>();
        //TODO: remember character with single quote
        myMap.put('{', '}');
        myMap.put('[', ']');
        myMap.put('(', ')');

        for (Character ch : testString.toCharArray()) {

            if (myMap.containsKey(ch)) {
                myStack.push(ch);
            } else if (myMap.containsValue(ch)) {

                if (myStack.isEmpty()) {
                    return false;
                }

                if (myMap.get(myStack.pop()) != ch) {
                    return false;
                }
            }
        }


        /*Iterator<Character> iter = myStack.iterator();

        while (iter.hasNext() ) {
            System.out.println(iter.next());
        }*/

        if (myStack.isEmpty()) {
            return true;
        } else {
            return false;
        }
    }

    public static void main(String[] args) {
        assert isValid("()") == true;

        assert isValid("()[]{}") == true;

        assert isValid("([") == false;

        assert isValid("{[]}") == true;
    }
}

File: leetcode/ValidParenth.js
let assert = require('assert');
/**
 * @param s {string}
 * @return {boolean}
 */
let isValid = function(s) {
    let myStack = [];
    let myHashMap = { };
    myHashMap['{'] = '}';
    myHashMap['('] = ')';
    myHashMap['['] = ']';

    for (let i = 0; i < s.length; i++) {

        let ch = s[i];

        //if key exist
        if (ch in myHashMap) {
            myStack.push(ch);
        }

        else {
            for (let key in myHashMap) {
                if (myHashMap[key] === ch) {
                    if (myStack.length === 0) {
                        return false;
                    }

                    if (myHashMap[myStack.pop()] != ch) {
                        return false;
                    }
                }
            }
        }

    }


    if (myStack.length == 0) {
        return true;
    } else {
        return false;
    }

}

assert.equal(isValid("{}"), true);

assert.equal(isValid("()[]{}"), true);

assert.equal(isValid("(["), false);

assert.equal(isValid("{[]}"), true);
